web/web.go

```diff
@@ -918,12 +919,15 @@ func (h *Handler) version(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *Handler) quit(w http.ResponseWriter, r *http.Request) {
-	select {
-	case <-h.quitCh:
+	var stopped bool
+	h.quitOnce.Do(func() {
+		stopped = true
+		close(h.quitCh)
+	})
+	if stopped {
```

Comment:
This should be the opposite

Reply:
I squashed my fix, is it OK?

A
```
if cond {
} else {
}
```

B
```
if !cond {
} else {
}
```

If B is preferred over A in this project, I will fix further.

Reply:
Should we rename stopped to closed, and move the print inside the Do () ? Why do we have if else after where a simple if would be enough? Also is it safe to write after closing?

Reply:
Fixed in https://github.com/prometheus/prometheus/pull/8242/commits/bc423903a5f59564215392576df82a48e09f2672